### PR TITLE
[DPE-3167] 5/edge: Correct retry before initialising users and update CI

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
## About

The goal of this pull request is to fix publish to charmcraft.

## Notes

To make CI pass this PR contains a fix for user creation: a retry is added to cluster users creation

### CODE REVIEW

File: .github/workflows/release_5_edge.yaml [hunk 1]
```diff
@@ -28,21 +28,22 @@ jobs:
...
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm_without_cache.yaml@v5
+    with:
+      charmcraft-snap-channel: "latest/edge"
+
+  release:
+      name: Release charm
+      needs:
+        - ci-tests
+        - build
+      uses: canonical/data-platform-workflows/.github/workflows/release_charm.yaml@v5
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply .github/workflows/release_5_edge.yaml:42]: Done

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        - ci-tests
+        - build
+      uses: canonical/data-platform-workflows/.github/workflows/release_charm.yaml@v5
+      with:
+        channel: 5/edge
+        artifact-name: ${{ needs.build.outputs.artifact-name }}
+      secrets:
+        charmhub-token: ${{ secrets.CHARMHUB_TOKEN }}
+      permissions:
+        contents: write  # Needed to create GitHub release
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply .github/workflows/release.yaml:49]: no, that was by mistake. Fixed

**Review by MiaAltieri — COMMENTED** [review]
looks good but +1 to @carlcsaposs-canonical comment

**Review by delgod — APPROVED** [review]

File: src/charm.py [hunk 3]
```diff
@@ -402,7 +406,11 @@ def _on_start(self, event) -> None:
             return
 
         self._initialise_replica_set(event)
-        self._initialise_users(event)
+        try:
+            self._initialise_users(event)
+        except RetryError:
+            event.defer()
```
Comment (MiaAltieri) [inline src/charm.py:412]: nit: add a log here

**Review by delgod — APPROVED** [review]

**Review by MiaAltieri — APPROVED** [review]

File: .github/workflows/release_5_edge.yaml [hunk 4]
```diff
@@ -28,21 +28,22 @@ jobs:
...
-          fetch-depth: 0
-      - name: Upload charm to charmhub
-        uses: canonical/charming-actions/upload-charm@2.3.0
-        with:
-          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
-          github-token: "${{ secrets.GITHUB_TOKEN }}"
-          channel: "5/edge"
+   build:
+    name: Build charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm_without_cache.yaml@v6
```
[the remark this answers is not in the document]
Comment (dmitry-ratushnyy) [reply .github/workflows/release_5_edge.yaml]: Done, thank you

File: src/charm.py (hunk 3 above)
Comment (dmitry-ratushnyy) [reply src/charm.py:412]: Done, thank you

**Pull request merged**